projects/opyn-zen-bull/index.js

```diff
@@ -1,71 +1,37 @@
 const EULER_SIMPLE_LENS_ADDRESS = "0x5077B7642abF198b4a5b7C4BdCE4f03016C7089C"
+const crab = "0x3B960E47784150F5a63777201ee2B15253D713e8";
 
 // to add TVL under yield category
-async function tvl(timestamp, block, _, { api }){
-    let balances = {};
-
-    // get eth usd price
-    const key = 'ethereum:0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2';      
-    const ethPriceUSD = (await post("https://coins.llama.fi/prices", {
-        "coins": [key]
-    })).coins[key].price;
-    // get eth usd price
-    const squeethKey = 'ethereum:0xf1b99e3e573a1a9c5e6b2ce818b617f0e664e86b';
-    const squeethPriceUSD = (await post("https://coins.llama.fi/prices", {
-        "coins": [squeethKey]
-    })).coins[squeethKey].price;
-
-    const [ethInCrab, squeethInCrab] = (await sdk.api.abi.call({
-        target: ZEN_BULL_ADDRESS,
-        abi: zenBullAbi.find(({ name }) => name === 'getCrabVaultDetails'),
-        chain: "ethereum"
-    })).output;
-    const bullCrabBalance = (await sdk.api.abi.call({
-        target: ZEN_BULL_ADDRESS,
-        abi: zenBullAbi.find(({ name }) => name === 'getCrabBalance'),
-        chain: "ethereum"
-    })).output;
-    const crab = "0x3B960E47784150F5a63777201ee2B15253D713e8";
-    const crabTotalSupply = (await sdk.api.erc20.totalSupply({
-        target: crab,
-        chain: "ethereum"
-      })
-    ).output
-    const bullDtokenBalance = (await sdk.api.abi.call({
-        target: EULER_SIMPLE_LENS_ADDRESS,
-        abi: eulerSimpleLens.find(({ name }) => name === 'getDTokenBalance'),
-        params: [USDC, ZEN_BULL_ADDRESS],
-        chain: "ethereum"
-    })).output;
-    const bullEtokenBalance = (await sdk.api.abi.call({
-        target: EULER_SIMPLE_LENS_ADDRESS,
-        abi: eulerSimpleLens.find(({ name }) => name === 'getETokenBalance'),
-        params: [WETH, ZEN_BULL_ADDRESS],
-        chain: "ethereum"
-    })).output;
-
-    const crabEthPrice = (ethInCrab  - (squeethInCrab * (squeethPriceUSD / ethPriceUSD ))) * 1e18 / crabTotalSupply;    
-    const balance = BigNumber(bullEtokenBalance).plus(BigNumber(bullCrabBalance).times(crabEthPrice).dividedBy(1e18)).minus(bullDtokenBalance / 1e6 / ethPriceUSD * 1e18)
-    
-    balances = {
-        [ETH]: BigNumber(balances[ETH] || 0).plus(balance).toFixed(0),
-    };
-
-    return balances;
+async function tvl(timestamp, block, _, { api }) {
+  let balances = {};
+
+  const bullDtokenBalance = await api.call({
+    target: EULER_SIMPLE_LENS_ADDRESS,
+    abi: eulerSimpleLens.getDTokenBalance,
+    params: [USDC, ZEN_BULL_ADDRESS],
+  })
+  const bullEtokenBalance = await api.call({
+    target: EULER_SIMPLE_LENS_ADDRESS,
+    abi: eulerSimpleLens.getETokenBalance,
+    params: [WETH, ZEN_BULL_ADDRESS],
+  })
+
+  sdk.util.sumSingleBalance(balances, WETH, bullEtokenBalance)
+  sdk.util.sumSingleBalance(balances, USDC, bullDtokenBalance)
+  return sumTokens2({ balances, api, owner: ZEN_BULL_ADDRESS, tokens: [crab] })
```

Comment:
So those 3 lines do the same as this?

```
    const crabEthPrice = (ethInCrab  - (squeethInCrab * (squeethPriceUSD / ethPriceUSD ))) * 1e18 / crabTotalSupply;    
    const balance = BigNumber(bullEtokenBalance).plus(BigNumber(bullCrabBalance).times(crabEthPrice).dividedBy(1e18)).minus(bullDtokenBalance / 1e6 / ethPriceUSD * 1e18)

    balances = {
        [ETH]: BigNumber(balances[ETH] || 0).plus(balance).toFixed(0),
    };
    ```

Reply:
yup, would return a result like 

    --- ethereum ---
    WETH                      762.46 k
    USDC                      322.57 k
    Crabv2                    320.21 k
    Total: 1.41 M 

    --- tvl ---
    WETH                      762.46 k
    USDC                      322.57 k
    Crabv2                    320.21 k
    Total: 1.41 M 

    ------ TVL ------
    ethereum                  1.41 M

    total                    1.41 M